[FangRuoqing] iP

FangRuoqing opened the pull request:
# RochinBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

RochinBot is a **task management application** designed to help you stay organized and productive. It offers a range of features to manage your tasks effectively.

## FEATURES:
* Add tasks
* Delete tasks
* Mark tasks as done
* Search for tasks by keyword

## Getting Started:
1. Download the RochinOOP.jar file from [here](https://github.com/FangRuoqing/ip).
2. Double-click the downloaded file to launch the application.

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class RochinOOP {
    public static void main(String[] args) {
        new RochinOOP().run();
    }
}
```
## Tasks to do:
- [x] Implement basic task management functionality
- [ ] Add support for managing deadlines
- [ ] Incorporate reminders for upcoming tasks

RochinOOP helps you keep track of your tasks, freeing your mind to focus on what truly matters. It's fast, easy to use, and completely free!

***So, let RochinBot handle your tasks for you!*** 🚀

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

FangRuoqing committed:
Add Rochin.java

FangRuoqing committed:
Level 1 Echo

FangRuoqing committed:
Level 2 Add, List

FangRuoqing committed:
Level 3 Mark as Done

FangRuoqing committed:
Level 4 ToDos, Events, Deadlines

FangRuoqing committed:
A-TextUiTesting: Automated Text UI Testing

FangRuoqing committed:
Level 5 Handle Errors

FangRuoqing committed:
Level 6 Delete

FangRuoqing committed:
Add new task

FangRuoqing committed:
Add new function level 7 Save

FangRuoqing committed:
Add javadoc

FangRuoqing committed:
modify javadoc

FangRuoqing committed:
Merge branch 'branch-Level-7'

FangRuoqing committed:
teach the chatbot how to understand dates and times. level 8

FangRuoqing committed:
update the data file

FangRuoqing committed:
Add more javadoc

FangRuoqing committed:
Merge branch 'master' into branch-Level-8

FangRuoqing committed:
Add more OOP

FangRuoqing committed:
Divide classes into packages

FangRuoqing committed:
Merge branch 'add-gradle-support'

FangRuoqing committed:
Add JUnit tests to test the behavior of the code.

FangRuoqing committed:
Package the App as a JAR file

FangRuoqing committed:
Add JavaDoc comments to the code.

FangRuoqing committed:
Tweak the code to comply with a coding standard

FangRuoqing committed:
Add more commentrs

FangRuoqing committed:
Merge branch 'branch-A-CodingStandard'

FangRuoqing committed:
Give users a way to find a task by searching for a keyword.

FangRuoqing committed:
add comments

FangRuoqing committed:
Modify comments

FangRuoqing committed:
Merge branch 'branch-Level-9'

FangRuoqing committed:
Merge branch 'branch-Level-8'

FangRuoqing committed:
Modify comments

FangRuoqing committed:
Modify more comments

tanjieling0 commented on src/main/java/Rochin.java:
```diff
@@ -0,0 +1,585 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.io.*;
```
Perhaps you could import individually to adhere to the coding standard?

tanjieling0 commented on src/main/java/Rochin.java:
```diff
@@ -0,0 +1,585 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    private void processCommands() throws RochinException {
```
Maybe processCommand(), without the plurality would be more fit for the method name?

tanjieling0 commented on src/main/java/Rochin.java:
```diff
@@ -0,0 +1,585 @@
+    /**
+     * Represent a command processor for processing user commands and managing tasks.
+     */
+    private class CommandProcessor {
```
Perhaps packaging the different classes into different packages/files would be more OOP?

tanjieling0 commented on src/main/java/Rochin.java:
```diff
@@ -0,0 +1,585 @@
+/**
+ * Represent a task with a description and a status.
+ */
+class Task {
```
Similar to above, maybe it would be cleaner to place the different classes into different files instead of condensing it into a single file?

tanjieling0 commented on src/main/java/rochin/RochinOOP.java:
```diff
@@ -0,0 +1,859 @@
+package rochin;
+
+import java.io.*;
```
Similar to above, perhaps you could import individually?

tanjieling0 commented on src/main/java/rochin/RochinOOP.java:
```diff
@@ -0,0 +1,859 @@
+    /**
+     * Method to process user commands.
+     */
+    public void processCommands() {
```
Is this the same method as in Cochin.java?

tanjieling0 commented on src/main/java/rochin/RochinOOP.java:
```diff
@@ -0,0 +1,859 @@
+         * @param keyword The keyword to search for in task descriptions.
+         * @return A list of tasks that match the given keyword.
+         */
+        public List<Task> findTasks(String keyword) {
```
Perhaps removing the plurality will be more fit?

tanjieling0 reviewed (commented):
Overall, I think that practicing more OOP principles, separating the codes into different files and deleting duplicate codes could make your code cleaner and more concise :)

FangRuoqing committed:
Add RochinGUI, Launcher, DialogBox

FangRuoqing committed:
Add some comments

FangRuoqing committed:
Merge branch 'master' into branch-Level-10

FangRuoqing committed:
Modify RochinGUI

FangRuoqing committed:
Change some comments

FangRuoqing committed:
Merge branch 'master' into branch-Level-10

FangRuoqing committed:
Add more comments

FangRuoqing committed:
Merge branch 'master' into branch-Level-10

FangRuoqing committed:
Change some comments

FangRuoqing committed:
Merge branch 'branch-Level-10'

FangRuoqing committed:
Rewrite RochinOOP by seperating classes

Previously all the classes are in the class RochinOOP.

Now I am trying to rewrite the code.

FangRuoqing committed:
Modify the rewritten classes

Some methods are not implemented correctly in the previous version

Have changed them to satisfy the requirements

FangRuoqing committed:
Improve code quality

FangRuoqing committed:
Add assert features

FangRuoqing committed:
Merge pull request #2 from FangRuoqing/branch-A-Assertions

Add assert features

FangRuoqing committed:
Merge pull request #1 from FangRuoqing/branch-A-CodeQuality

Improve code quality

FangRuoqing committed:
Update the code accordingly

FangRuoqing committed:
Fix save data bug

FangRuoqing committed:
Modify some methods

FangRuoqing committed:
Merge commit '2714e8eaa501d3275d93fee613a831be7d3726cd'

FangRuoqing committed:
Add some comments

FangRuoqing committed:
Modify GUI

FangRuoqing committed:
Modify more GUI

FangRuoqing committed:
Add more GUI

FangRuoqing committed:
Change the pictures of GUI

FangRuoqing committed:
Fix the bug of creating a new file

FangRuoqing committed:
Update User Guide

FangRuoqing committed:
Update java docs

FangRuoqing committed:
Add user help page
